### PR TITLE
Remove the logic of padding head dim to 128 before calling to kernel.

### PR DESCRIPTION
**Pull request by QiliangCui**
# Description

PRA3 kernel natively supports head dim being less than 128. So, we can remove the padding logic.

FIXES: b/440571948

# Tests
1. Ran on benchmark testing machines and got successful run.
2. Add a CIT to cover head-dim not equal to 128. (https://buildkite.com/tpu-commons/tpu-commons-ci/builds/2176)

# Checklist

Before submitting this PR, please make sure:
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tests/kernels/ragged_paged_attention_kernel_v3_test.py [hunk 1]
```diff
@@ -498,6 +499,24 @@ def test_ragged_paged_attention_soft_cap_cannot_be_zero(self):
                 soft_cap=0.0,
             )
 
+    @parameterized.product(dtype=[jnp.float32, jnp.bfloat16],
+                           head_dim=[64, 156, 256])
+    def test_ragged_paged_attention_padding(self, dtype, head_dim):
```
Comment (bythew3i) [inline tests/kernels/ragged_paged_attention_kernel_v3_test.py]: The `test_ragged_paged_attention_complex` should already cover unaligned head_dims. I would prefer not adding this test since the RPA tests are very slow due to the reference RPA. WDYT?
